### DISCUSSION TITLE
Where does `action/upload-pages` place built files?

### DISCUSSION THREAD

**Discussion by revosw** [community/community#28253] [Other Feature Feedback, Questions, & Ideas]
I am trying to use [VitePress](https://vitepress.vuejs.org/) as a static site generator for a [documentation site I'm making](https://github.com/revosw/learnwiremod).

## Background
I saw a video demonstrating [publishing a github pages site using Github Actions](https://www.youtube.com/watch?v=Z_7RIuf_Z-Q). The entire process seem to rely on two actions.

The `actions/upload-pages-artifact@v1` action takes the path to the built files as input. It first compresses the folder and its contents into a tar file, and then uploads this artifact.

The `actions/deploy-pages@main` takes the `github-pages` artifact created by the upload-pages-artifact action, and creates an environment also called `github-pages`. From here, the rest is just magic to me.

### How to build with VitePress
To build with VitePress, make a `docs` folder in the root directory with markdown files inside. Then run `npx vitepress build docs` to transform all the markdown and .vue files into static html. These files are placed inside `docs/.vitepress/`. [The workflow file](https://github.com/revosw/learnwiremod/actions/runs/2791858182/workflow) shows that the `upload-pages-artifact` uploads the `./docs/.vitepress` folder. The folder is avilable to [download as an artifact](https://github.com/revosw/learnwiremod/actions/runs/2791858182)

## Problem
I expected [my workflow](https://github.com/revosw/learnwiremod/actions/runs/2791858182/workflow) to properly set up github pages from the built files. As far as I can see from [the workflow log](https://github.com/revosw/learnwiremod/runs/7659244368?check_suite_focus=true), everything seems fine! However, all I get is a 404 page. I have the feeling that the static content is there somewhere, but it's hidden in some `/build/` path or something that's impossible to guess. How can I ensure that the built files lie at the root of my website? I've ruled out DNS problems by removing the custom domain name from the github pages settings page. (don't think I would have reached the 404 page at all if _that_ was a problem)

Just for completeness' sake, I have of course set build and deployment source to github actions
![image](https://user-images.githubusercontent.com/19785016/182711083-cfa8aba8-0391-49db-90f2-ee0c7467e20c.png)

## An observation after posting it originally
There seems to be a [third action](https://github.com/actions/configure-pages) related to this, `action/configure-pages@v1`. It looks like VitePress may be implemented alongside gatsby, nuxt and next

**Comment by revosw** [comment]
@yoannchaudet I'd be happy to contribute if there's anything VitePress specific I must implement in any of the actions to make it up-to-par with other supported SSGs. I'd need some guidance on that, though I imagine

**Comment by yoannchaudet** [comment]
@revosw Your workflow looks fine. You can even check the artifacts it is uploading by checking the "Summary" view and downloading your artifacts:

![image](https://user-images.githubusercontent.com/14911070/182889051-a630c623-1f94-479d-863c-85ca1bebfe0f.png)

Your issue here is not your workflow. Your Pages site is configured to track your `master` branch. Any push you make, you have two workflows competing to deploy your site: https://github.com/revosw/learnwiremod/actions/workflows/pages/pages-build-deployment.

Go to your Pages settings, and select "GitHub Actions" as the source. The automatic workflows will stop being triggered and only your custom workflow will manage your stie. You can learn more about that here: https://github.blog/changelog/2022-07-27-github-pages-custom-github-actions-workflows-beta/

**Reply by revosw** [reply]
That's what I find strange, the artifacts are fine. I did put GitHub Actions as my source yesterday. I have now additionally unpublished from the Pages section in settings, and deleted the `gh-pages` branch, then re-ran the build and deploy jobs in my workflow which re-publishes. Still 404

**Reply by yoannchaudet** [reply]
Like I said, if you re-build your site, download the artifacts you will realize that you are publishing a whole `dist` folder instead of just the content of this folder.

So your site lives at https://learnwiremod.com/dist/. You will need to tweak your [workflow a bit](https://github.com/revosw/learnwiremod/blob/e2ae7c7be1c2f0438ccc97a040e8a1ba0d7e1fd0/.github/workflows/deploy.yml#L32).

**Reply by revosw** [reply]
Thank you! I gave up on it shortly after, but now I finally understand.
